src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,463 @@
+      (event: PointerEvent) => {
+        updateThumbValue(event, onChange);
+        setActiveThumbIndex(undefined);
+        event.preventDefault();
+        event.stopPropagation();
+      },
+      [onChange, updateThumbValue],
+    );
+
+    const handlePointerDownOnSlider = React.useCallback(
+      (event: React.PointerEvent) => {
+        if (containerRef.current) {
+          const percent = getPercentageOfRectangle(
+            containerRef.current.getBoundingClientRect(),
+            event.pageX,
+          );
+          let pointerValue = min + (max - min) * percent;
+          pointerValue = roundValueToClosestStep(pointerValue, step, min);
+
+          const closestValueIndex = getClosestValueIndex(
+            currentValues,
+            pointerValue,
+          );
+          const [minVal, maxVal] = getAllowableThumbRange(closestValueIndex);
+          pointerValue = getBoundedValue(pointerValue, minVal, maxVal);
+          if (pointerValue === currentValues[closestValueIndex]) {
+            return;
+          }
+          const newValues = [...currentValues];
+          newValues[closestValueIndex] = pointerValue;
+          setCurrentValues(newValues);
+          onChange?.(newValues);
+        }
+      },
+      [min, max, step, currentValues, getAllowableThumbRange, onChange],
+    );
+
+    const pointerMoveFunctionRef = React.useRef<
+      (event: PointerEvent) => void
+    >();
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      if (pointerMoveFunctionRef.current && ownerDoc) {
+        ownerDoc.removeEventListener(
+          'pointermove',
+          pointerMoveFunctionRef.current,
+        );
+      }
+      pointerMoveFunctionRef.current = handlePointerMove;
+    }, [handlePointerMove]);
+
+    const pointerUpFunctionRef = React.useRef<(event: PointerEvent) => void>();
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      if (pointerUpFunctionRef.current && ownerDoc) {
+        ownerDoc.removeEventListener('pointerup', pointerUpFunctionRef.current);
+      }
+      pointerUpFunctionRef.current = handlePointerUp;
+    }, [handlePointerUp]);
+
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      ownerDoc?.addEventListener('pointermove', handlePointerMove);
+      ownerDoc?.addEventListener('pointerup', handlePointerUp);
+      return () => {
+        ownerDoc?.removeEventListener('pointermove', handlePointerMove);
+        ownerDoc?.removeEventListener('pointerup', handlePointerUp);
+      };
+    }, [handlePointerMove, handlePointerUp]);
+
+    return (
+      <div
+        className={cx(
+          'iui-slider-component-container',
+          disabled && 'iui-disabled',
+          className,
+        )}
+        {...rest}
+      >
+        {minValueLabel && (
+          <span className='iui-slider-min'>{minValueLabel}</span>
+        )}
+        <div
+          ref={refs}
```

Comment:
Any reason user-provided ref needs to be set here?

Reply:
Well I need this ref and not the root one.... but you are correct, I probably need to return the root one.